Setting.clear_cache!

linki opened the pull request:
- clears all cached settings

todos, that came up:
- check how it behaves with cache namespaces (we want to scope everything by namespace): `delete_matched` will respect the namespace, i.e. don't delete anything in another namespace. `clear` clears all namespaces.
- `delete_matched` isn't even implemented by `DalliStore`, so maybe we shouldn't fall back to `clear` that early
- we also need to rescue from `NoMethodError` because implementations not necessarily descend from the base class

Commit:
add Setting.clear_cache!

* clears all cached settings

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://linki-feature-clear-settings-cache.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://linki-feature-clear-settings-cache.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

linki commented on spec/models/setting_spec.rb:
```diff
@@ -169,4 +169,40 @@
+      expect(Rails.cache.read('foo')).to eq 'bar'
+    end
+
+    it "falls back to clearing everything if partial deletion is unsupported" do
```
switching between single and double-quotes based on the existence of a single-quote in the description itself is just wrong, imo. look at the test above which contains `'`.

linki commented on app/models/setting.rb:
```diff
@@ -201,6 +201,15 @@ def self.find_or_default(name)
+  # cache. if #clear is also not supported we just yawn and fail together.
+  def self.clear_cache!
+    Rails.cache.delete_matched /\/openproject\/settings\/.*/
+  rescue NotImplementedError
```
`rescue NotImplementedError, NoMethodError`

we also need to rescue from `NoMethodError` because implementations not necessarily descend from the base class.

Commit:
rescue from NoMethodError as well

implementations not necessarily descend from the
base class. then you'll get a NoMethodError
instead of NotImplementedError

basiszwo commented:
closing this as it obviously wasn't important enough to be finished.
